lib/replay_api/client.rb

```diff
@@ -6,37 +6,25 @@
 module ReplayApi
   class Client
     attr_reader :configuration
-    attr_writer :connection
+    attr_accessor :deliverer
 
-    def initialize(configuration=ReplayApi.configuration)
+    def initialize(configuration=ReplayApi.configuration, deliverer=ReplayApi::RequestDeliverer.instance)
       @configuration ||= configuration
+      @deliverer ||= deliverer
     end
 
     def event(event=Event.new)
+      deliverer.post(uri: '/events', payload: payload(event.compact_attributes))
       yield event if block_given?
-      connection.post '/events', payload(event.compact_attributes)
     end
```

Comment:
You changed this from posting after yield to posting before yield which means people using the block form can't fiddle with the attributes.